libbeat/publisher/queue/diskqueue/benchmark_test.go

```diff
@@ -0,0 +1,133 @@
+		"2 123456789010 eni-1235b8ca123456789 - - - - - - - 1431280876 1431280934 - NODATA",
+		"Oct 10 2018 12:34:56 localhost CiscoASA[999]: %ASA-6-305012: Teardown dynamic TCP translation from inside:172.31.98.44/1459 to outside:192.168.98.44/8272 duration 0:00:30",
+		"{\"timestamp\":\"2021-01-27T01:28:11.488362+0100\",\"flow_id\":1805461738637437,\"in_iface\":\"enp6s0\",\"event_type\":\"alert\",\"src_ip\":\"175.16.199.1\",\"src_port\":80,\"dest_ip\":\"10.31.64.240\",\"dest_port\":47592,\"proto\":\"TCP\",\"ether\":{\"src_mac\":\"00:03:2d:3f:e5:63\",\"dest_mac\":\"00:1b:17:00:01:18\"},\"alert\":{\"action\":\"allowed\",\"gid\":1,\"signature_id\":2100498,\"rev\":7,\"signature\":\"GPL ATTACK_RESPONSE id check returned root\",\"category\":\"Potentially Bad Traffic\",\"severity\":2,\"metadata\":{\"created_at\":[\"2010_09_23\"],\"updated_at\":[\"2010_09_23\"]}},\"http\":{\"hostname\":\"testmynids.org\",\"url\":\"/uid/index.html\",\"http_user_agent\":\"curl/7.58.0\",\"http_content_type\":\"text/html\",\"http_method\":\"GET\",\"protocol\":\"HTTP/1.1\",\"status\":200,\"length\":39},\"app_proto\":\"http\",\"flow\":{\"pkts_toserver\":6,\"pkts_toclient\":5,\"bytes_toserver\":496,\"bytes_toclient\":876,\"start\":\"2021-01-22T23:28:38.673917+0100\"}}",
+		"2018-08-28 18:24:25 [10.100.220.70](http://10.100.220.70) GET / - 80 - [10.100.118.31](http://10.100.118.31) Mozilla/4.0+(compatible;+MSIE+7.0;+Windows+NT+6.3;+WOW64;+Trident/7.0;+.NET4.0E;+.NET4.0C;+.NET+CLR+3.5.30729;+.NET+CLR[+2.0.50727](tel:+2050727);+.NET+CLR+3.0.30729) 404 4 2 792",
+	}
+)
+
+//makeEvent creates a sample event, using a random message from msg above
+func makeEvent() publisher.Event {
+	return publisher.Event{
+		Content: beat.Event{
+			Timestamp: eventTime,
+			Fields: mapstr.M{
+				"message": msgs[rand.Intn(len(msgs))],
+			},
+		},
+	}
+}
+
+//setup creates the disk queue, you must remove the directory it makes
+func setup() (*diskQueue, queue.Producer, queue.Consumer) {
+	dir, err := os.MkdirTemp("", "benchmark")
+	if err != nil {
+		panic(err)
+	}
+	s := DefaultSettings()
+	s.Path = dir
+	q, err := NewQueue(logp.NewLogger("benchmark"), s)
+	if err != nil {
+		os.RemoveAll(dir)
+		panic(err)
+	}
+	prod := q.Producer(queue.ProducerConfig{})
+	cons := q.Consumer()
+	return q, prod, cons
+}
+
+//clean closes the queue and deletes the directory
+func cleanup(q *diskQueue) {
+	q.Close()
+	os.RemoveAll(q.settings.directoryPath())
```

Comment:
Wait, is this the temp path created in `setup` ? If so , might want to clarify the comment above the function.

Reply:
yep, I updated the comments.  let me know if that helps clear it up.  Basically the queue has to live on disk somewhere and we need to delete it when we are done.